upgrade/1.5.1/README.md

```diff
@@ -0,0 +1,312 @@
+and exits with status code `0`.
+
+### Update Argo CRDs
+
+(`ncn-m001#`) Run the following script in preparation for 1.5.1 patch upgrade:
+
+```bash
+for c in $(kubectl get crd |grep argo | cut -d' ' -f1)
+do
+   kubectl label --overwrite crd $c app.kubernetes.io/managed-by="Helm"
+   kubectl annotate --overwrite crd $c meta.helm.sh/release-name="cray-nls"
+   kubectl annotate --overwrite crd $c meta.helm.sh/release-namespace="argo"
+done
+```
+
+### Upgrade services
+
+(`ncn-m001#`) Run `upgrade.sh` to deploy upgraded CSM applications and services:
+
+```bash
+cd "$CSM_DISTDIR"
+./upgrade.sh
+```
+
+### Upload NCN images
+
+It is important to upload NCN images to IMS and to edit the `cray-product-catalog`. This is necessary when updating
+products with IUF. If this step is skipped, IUF will fail when updating or upgrading products in the future.
+
+(`ncn-m001#`) Execute script to upload CSM NCN images and update the `cray-product-catalog`.
+
+```bash
+/usr/share/doc/csm/upgrade/scripts/upgrade/upload-ncn-images.sh
+```
+
+### Upgrade Ceph and stop local Docker registries
+
+**Note:** This step may not be necessary if it was already completed by the CSM `v1.3.5` patch or `v1.4.2` patch. If it was already run,
+the following steps can be re-executed to verify that Ceph daemons are using images in Nexus and the local Docker
+registries have been stopped.
+
+These steps will upgrade Ceph to `v16.2.13`. Then the Ceph monitoring daemons' images will be pushed to Nexus and the
+monitoring daemons will be redeployed so that they use these images in Nexus. Once this is complete, all Ceph daemons
+should be using images in Nexus and not images hosted in the local Docker registry on storage nodes. The third step
+stops the local Docker registry on all storage nodes.
+
+1. (`ncn-m001#`) Run Ceph upgrade to `v16.2.13`.
+
+   ```bash
+   /usr/share/doc/csm/upgrade/scripts/ceph/ceph-upgrade-tool.py --version "v16.2.13"
+   ```
+
+1. (`ncn-m001#`) Redeploy Ceph monitoring daemons so they are using images in Nexus.
+
+   ```bash
+   scp /usr/share/doc/csm/scripts/operations/ceph/redeploy_monitoring_stack_to_nexus.sh ncn-s001:/srv/cray/scripts/common/redeploy_monitoring_stack_to_nexus.sh
+   ssh ncn-s001 "/srv/cray/scripts/common/redeploy_monitoring_stack_to_nexus.sh"
+   ```
+
+1. (`ncn-m001#`) Stop the local Docker registries on all storage nodes.
+
+   ```bash
+   scp /usr/share/doc/csm/scripts/operations/ceph/disable_local_registry.sh ncn-s001:/srv/cray/scripts/common/disable_local_registry.sh
+   ssh ncn-s001 "/srv/cray/scripts/common/disable_local_registry.sh"
+   ```
+
+### Enable `smartmon` metrics on storage NCNs
+
+This step will install the `smart-mon` rpm on storage nodes, and reconfigure the `node-exporter` to provide `smartmon`
+metrics.
+
+1. (`ncn-m001#`) Execute the following script.
+
+   ```bash
+   /usr/share/doc/csm/scripts/operations/ceph/enable-smart-mon-storage-nodes.sh
+   ```
+
```

Comment:
```suggestion
```

Remove the entire blocks 'upgrade ceph and stop local docker registries' and 'Enable smartmon on storage nodes'. Both of these are fixed in CSM 1.5 and don't need to be in the patch instructions